config/file.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Stefan Prodan 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package config
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"gopkg.in/yaml.v2"
+)
+
+var (
```

Comment:
Out of curiosity.. is this all original work, or based upon an example?

Reply:
The naming convention follows docker/cli/config and the decode function is using the same escape as in docker registry store, other than that is as "original" as this little pice of code can be :)

Reply:
Can you link me to that code please? Thank you

Reply:
@alexellis do you still need this?

Reply:
Sorry I've missed this question. Here it is https://github.com/docker/cli/blob/master/cli/config/configfile/file.go